tests/Tests.AzureAppConfiguration/RefreshTests.cs

```diff
@@ -559,7 +510,7 @@ public void RefreshTests_RefreshAsyncThrowsOnRequestFailedException()
                     options.Select("TestKey*");
                     options.ConfigureRefresh(refreshOptions =>
                     {
-                        refreshOptions.Register("TestKey1")
+                        refreshOptions.Register("NonExistentKey") // Exception will be thrown only if a non-existent key is registered for refresh
```

Comment:
>Exception will be thrown only if a non-existent key is registered for refresh [](start = 69, length = 77)

I don't think it matters what key is used here. if we have configured the mock to throw an exception, it will be thrown. The comment could be confusing.

Reply:
In `RefreshIndividualKeyValues`, we check if the refresh registered KV exists in _settings cache. 

If it exists, we will call `GetKeyValueChange` which calls `GetConfigurationSettingAsync(ConfigurationSetting, bool, CancellationToken))`. 

If it doesn't exist, we call `GetConfigurationSettingAsync(string, string, CancellationToken))` - which is the method configured to throw the exception.

So we have two options - either register a non existent key or setup a different method to throw the exception. Since it doesn't really matter which method throws exception for the purpose of these two tests, I just changed the key to be explicitly non existent.

Reply:
I prefer the former approach over the latter for three reasons.

1. A user is more likely to use a key that exists as compared to one that does not.
2. Using a non-existent key distracts from the original test case that is being validated and is not necessary.
3. Implementation that requires no comments is generally easier to understand than one that requires a comment (despite the explanation).

---
In reply to: [503572923](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/205#discussion_r503572923) [](ancestors = 503572923)

Reply:
Makes sense. Updated both tests.